src/remoteConnector.ts

```diff
@@ -806,8 +807,22 @@ export default class RemoteConnector extends Disposable {
 				this.telemetry.sendRawTelemetryEvent('vscode_desktop_ssh', { kind: 'gateway', status: 'failed', reason: e.toString(), ...params, gitpodVersion: gitpodVersion.raw, userOverride });
 				if (e instanceof NoSSHGatewayError) {
 					this.logger.error('No SSH gateway:', e);
-					vscode.window.showWarningMessage(`${e.host} does not support [direct SSH access](https://github.com/gitpod-io/gitpod/blob/main/install/installer/docs/workspace-ssh-access.md), connecting via the deprecated SSH tunnel over WebSocket.`);
-					// Do nothing and continue execution
+					const ok = 'Ok';
+					const action = await vscode.window.showWarningMessage(`${e.host} does not support [direct SSH access](https://github.com/gitpod-io/gitpod/blob/main/install/installer/docs/workspace-ssh-access.md), connecting via the deprecated SSH tunnel over WebSocket.`, ok);
+					if (action === ok) {
+						// Do nothing and continue execution
+					} else {
+						return;
+					}
+				} else if (e instanceof SSHError && e.message === 'Timed out while waiting for handshake') {
+					this.logger.error('SSH test connection error:', e);
+					const ok = 'Ok';
+					const action = await vscode.window.showWarningMessage(`Timed out while waiting for SSH handshake, it's possible SSH connections on port 22 are blocked or your network is too slow, connecting via the deprecated SSH tunnel over WebSocket.`, ok);
```

Comment:
cc @akosyakov if you want to tweak the message

Reply:
it looks good to me

cc @loujaybee

Reply:
from @filiptronicek https://github.com/gitpod-io/gitpod-vscode-desktop/pull/20#discussion_r959299375